src/subapps/search/hooks/useGlobalSearch.tsx

```diff
@@ -65,6 +65,70 @@ function filterReducer(
   }
 }
 
+function fieldVisibilityReducer(
```

Comment:
local storage change is technically a side effect like dom updates, network calls etc. Instead of doing this in reducer, we can move this to a useEffect that just handles all local storage updates.

useffect(() => {
      localStorage.setItem(
        'search-field-visibility',
        JSON.stringify(FieldsVisibilityState) 
      ); // Something of this sort.
}, [FieldsVisibilityState])

We don't need to do this in this PR, can create a tech debt issue and address this late.